### PR TITLE
HPCC-22029 Return DFU file type in text response

### PR DESCRIPTION
**Pull request by jakesmith**
Also add locking timeout to request (optional),
and sessionId (future use).

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@rpastrana - please review/test.

File: esp/scm/ws_dfu.ecm [hunk 1]
```diff
@@ -861,6 +861,8 @@ ESPrequest DFUFileAccessV2Request
     string RequestId;                   // optional
     int ExpirySeconds(60);              // how long the retrieved meta info is valid for, to access physical files (in dafilesrv)
     bool ReturnTextResponse(false);     // Returns response in full not just in blob    
+    [min_ver("1.51")] int64 SessionId(0); // session id of this client request    
+    [min_ver("1.51")] int LockTimeoutMs(300000); // Max seconds to block waiting for lock (default = 5 mins)
```
Comment (rpastrana) [inline esp/scm/ws_dfu.ecm]: comment claims timeout is in seconds, but its effectively a millisec timeout

File: esp/scm/ws_dfu.ecm [hunk 2]
```diff
@@ -936,7 +938,9 @@ ESPrequest [nil_remove] DFUFileCreateV2Request
     string RequestId;                   // optional 
     int ExpirySeconds(60);              // how long the retrieved meta info is valid for, to access physical files (in dafilesrv) 
     bool ReturnTextResponse(false);     // Returns response in full not just in blob
-    bool Compressed(false);             // The new file will be compressed on disk    
+    bool Compressed(false);             // The new file will be compressed on disk
+    [min_ver("1.51")] int64 SessionId(0); // session id of this client request    
+    [min_ver("1.51")] int LockTimeoutMs(300); // Max seconds to block waiting for lock
```
Comment (rpastrana) [inline esp/scm/ws_dfu.ecm]: this default lock timeout seems small just checking that it is intentional.

File: esp/services/ws_dfu/ws_dfuService.cpp [hunk 3]
```diff
@@ -6019,7 +6019,7 @@ SecAccessFlags translateToSecAccessFlags(CSecAccessType from)
     }
 }
 
-void CWsDfuEx::dFUFileAccessCommon(IEspContext &context, const CDfsLogicalFileName &lfn, const char *requestId, unsigned expirySecs, bool returnTextResponse, IEspDFUFileAccessInfo &accessInfo)
+void CWsDfuEx::dFUFileAccessCommon(IEspContext &context, const CDfsLogicalFileName &lfn, SessionId clientSessionId, const char *requestId, unsigned expirySecs, bool returnTextResponse, unsigned lockTimeoutMs, IEspDFUFileAccessResponse &resp)
```
Comment (rpastrana) [inline esp/services/ws_dfu/ws_dfuService.cpp]: is the lockTimeoutMs param actually used ?

**Review by rpastrana — COMMENTED** [review]
@jakesmith looks good, just a couple of questions

File: esp/scm/ws_dfu.ecm (hunk 1 above)
Comment (jakesmith) [reply esp/scm/ws_dfu.ecm]: comment should say milliseconds, will change

File: esp/scm/ws_dfu.ecm (hunk 2 above)
Comment (jakesmith) [reply esp/scm/ws_dfu.ecm]: yes, default should be consistent and same as above, will fix.

File: esp/services/ws_dfu/ws_dfuService.cpp (hunk 3 above)
Comment (jakesmith) [reply esp/services/ws_dfu/ws_dfuService.cpp]: appears not, should be passed to lookup (as it is in publish), will fix.

**Comment by jakesmith** [comment]
@rpastrana - fixed issues above.

**Review by rpastrana — APPROVED** [review]
@jakesmith approved

**Comment by jakesmith** [comment]
Squashed.

@richardkchapman - ready to merge.

**Comment by richardkchapman** [comment]
Also cherry-picked to 7.2.14

**Pull request merged**